### DISCUSSION TITLE
Smart control a heating system with HA?

### DISCUSSION THREAD

**Discussion by swa72** [emsesp/EMS-ESP32#965] [General]
Hi,

now that I have EMS-ESP and HA running smoothly together, I can tune my heating circuits (adjusting the heating curve, playing around with the radiators and stuff). 

One major problem in my house are many windows, sunny days and overshooting temperatures. https://github.com/proddy/EMS-ESP/issues/524 tells me not to mess with the boiler directly as any program running on the thermostat (RC300) may override this. So basically, I was thinking along the following lines:

If weather forecast equals sunny for the next day, reduce offset temperature by X early next morning (say Y minutes before sunrise) and reset offset temperature in the evening (say, Z minutes before sunset). Given my Aqara temperature sensors in the room, I feedback that info into the system and "learn" X, Y and Z and adjust offset temperatures.

Normally, its a job for a TRV to close the valve. However, I observe the following: given that the boiler only knows outside temperature, it will fire even if the TRVs closes their valves. The consequence, boiler starts and stops. Repeat.

I'm sure that tado and others are using similar ideas. Anyone knows of something similar for HA or any other system?

**Comment by glitter-ball** [comment]
My system kind of gets round that by having weather comp with room influence. If the sun shines in the room and warms it, the room temp exceeds the setpoint and the boiler flow temperature is reduced by the thermostat to prevent overheating.

Can you add room influence to your thermostat?

**Comment by mvjt** [comment]
> My system kind of gets round that by having weather comp with room influence. If the sun shines in the room and warms it, the room temp exceeds the setpoint and the boiler flow temperature is reduced by the thermostat to prevent overheating.
> 
> Can you add room influence to your thermostat?

My experience is that it is too late when the room thermostat detects it. I have a light sensor in my room sensor. I've installed it where the first morning light hits the inside of the house. It helped. But for best performance weather forecasts must be used so you can pull the brakes even earlier  :-) Just my experience...

**Comment by swa72** [comment]
Yup, I agree. That is why I integrated a solar forecast into my heating
control.

Am Mo., 8. Aug. 2022 um 21:39 Uhr schrieb mvjt ***@***.***>:

> My system kind of gets round that by having weather comp with room
> influence. If the sun shines in the room and warms it, the room temp
> exceeds the setpoint and the boiler flow temperature is reduced by the
> thermostat to prevent overheating.
>
> Can you add room influence to your thermostat?
>
> My experience is that it is too late when the room thermostat detects it.
> I have a light sensor in my room sensor. I've installed it where the first
> morning light hits the inside of the house. It helped. But for best
> performance weather forecasts must be used so you can pull the brakes even
> earlier :-) Just my experience...
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/emsesp/EMS-ESP32/issues/144#issuecomment-1208532779>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEGI2EZLOAXIUPHJDSK4RBDVYFO6RANCNFSM5FI5BEGQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Stefan Waldherr
Phone +49-151-1131-6967
